tests/rest/client/v2_alpha/test_upgrade_room.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Half-Shot
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional
+
+from synapse.config.server import DEFAULT_ROOM_VERSION
+from synapse.rest import admin
+from synapse.rest.client.v1 import login, room
+from synapse.rest.client.v2_alpha import room_upgrade_rest_servlet
+
+from tests import unittest
+from tests.server import FakeChannel
+
+
+class UpgradeRoomTest(unittest.HomeserverTestCase):
```

Comment:
Heads up we do have room upgrade tests in SyTest already, some of which attempt to test weird power level configurations: https://github.com/matrix-org/sytest/pull/777

But yay for more tests (and understandable if you'd rather not write perl)!

Reply:
Well, those sytests don't seem to catch this case. I don't know if it makes sense to add these as sytests or unit tests, but I couldn't find any unit tests for room upgrade, which seems bad.

Reply:
I'll try to add some, but Perl is fighting my at the moment!